src/lib/autoinstall/widgets/storage/add_drive_button.rb

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "cwm/common_widgets"
+
+module Y2Autoinstallation
+  module Widgets
+    module Storage
+      # Button to add new drive sections
+      #
+      # This button allows to add new drive sections of different types
+      # (`CT_DISK`, `CT_LVM`, `CT_RAID`, etc.).
+      class AddDriveButton < CWM::MenuButton
+        include Yast::Logger
+
+        # Constructor
+        #
+        # @param controller [Y2Autoinstallation::StorageController] UI controller
+        def initialize(controller)
+          textdomain "autoinst"
+          super()
+          @controller = controller
+          self.handle_all_events = true
```

Comment:
This is actually missing feature in MenuButton ( it is not much used ). It should add to handle all ids from items like it is done e.g. in custom widget https://github.com/yast/yast-yast2/blob/master/library/cwm/src/lib/cwm/custom_widget.rb#L41 Feel free to fill issue or bug report.